### PR TITLE
hwdb: Fixed Incorrect touchpad dimensions on Thinkpad L14 Gen1 (#22793)

### LINKED ISSUE #22793: Incorrect touchpad dimensions on Thinkpad L14 Gen1
**systemd version the issue has been seen with**
 > systemd 249 (v249.9-1.fc35)

**Used distribution**
 > Fedora Workstation 35

**Linux kernel version used** (`uname -a`)

 > `Linux fedora 5.16.15-201.fc35.x86_64 #1 SMP PREEMPT Thu Mar 17 05:45:13 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
`

**CPU architecture issue was seen on**
 > x86-64

The touchpad on my laptop doesn't feel right, and especially annoying is the unreasonably fast scroll speed. Looking around on how to troubleshoot it I noticed that the output of `libinput list-devices` said 70x50mm, while the actual physical dimensions are 100x70mm.

Libinput documentation said that I needed to run `libinput measure touchpad-size`, try the suggested hwdb.d entry and submit a patch to systemd.  I don't know a thing about using git , so hopefully this is the right place to ask for help in order to get the fix into systemd.

I followed the instructions on https://github.com/systemd/systemd/blob/main/hwdb.d/60-evdev.hwdb and  now `libinput list-devices` says the dimensions are 100x71mm. I ran `libinput measure touchpad-size` thrice and it keeps insisting on the 71mm, while in reality it is more like 69.5mm, I don't know how to adjust it or if it's advisable to do so. 

Thus, the hwdb.d entry to be added is, as suggested by `libinput measure touchpad-size` : 

```
# Lenovo Thinkpad L14 Gen1 (AMD)
evdev:name:SynPS/2 Synaptics TouchPad:dmi:*svnLENOVO:*pvrThinkPadL14Gen1**
 EVDEV_ABS_00=::44
 EVDEV_ABS_01=::50
 EVDEV_ABS_35=::44
 EVDEV_ABS_36=::50
```

**Considerations:**

1. The dmi match may need tuning, I used the one provided by `libinput measure touchpad-size` as is.
2. While the actual physical dimensions of the touchpad are 100x70mm, or more like 100x69.5mm, the previous hwdb.d entry causes libinput list devices to report the dimensions as 100x71mm.

This is the first time I use github for anything so I apologize if I did anything wrong, hopefully you can call out any mistakes I had.

### PR DESCRIPTION
**Pull request by taniishkaaa**
FIxes #22793

### CODE REVIEW

**Review by yuwata — CHANGES_REQUESTED** [review]
Please add the rule to 60-evdev.hwdb, instead of 70-touchpad.hwdb.

**Comment by yuwata** [comment]
Also, please drop the commits in your previous PR and rebase the branch to the current git HEAD, that can be done something like the following:
```sh
$ git remote add upstream https://github.com/systemd/systemd
$ git fetch upstream main
$ git rebase upstream/main
```

**Comment by poettering** [comment]
yeah, please move the dimension data over, but leave the ID_INPUT_TOUCHPAD_INTEGRATION= field in the current hwdb file

File: hwdb.d/70-touchpad.hwdb [hunk 1]
```diff
@@ -56,6 +56,12 @@ touchpad:usb:v05acp0265:*
 touchpad:usb:v044ep1221:*
  ID_INPUT_TOUCHPAD_INTEGRATION=external
 
+###########################################################
+# Lenovo
+###########################################################
+# Thinkpad L14 Gen1 (AMD)
+ #ID_INPUT_TOUCHPAD_INTEGRATION=internal
+
```
Comment (yuwata) [inline hwdb.d/70-touchpad.hwdb]: All lines are commented out... Please drop the change in 70-touchpad.hwdb.

**Review by yuwata — CHANGES_REQUESTED** [review]

**Comment by taniishkaaa** [comment]
Closing this PR in favor of #26937

**Pull request closed without merging**